### PR TITLE
Feature/344 my resources admin page

### LINKED ISSUE #344: Add my resources page for admins
Add the following info:
- [x] Name of the resource with link to resource details page.
- [x] Amount of continuous unreserved time from now.
- [x] Details about the next reservation for the day:
  - reservation start and end time
  - name of the person who made the reservation
  - comments about the reservation

See page 18 from the mockups.

### PR DESCRIPTION
**Pull request by YoYuUm**
Closes #344

### CODE REVIEW

File: app/components/resource/ResourcesTableItem.js [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+      currentReservation,
+      nextReservation,
+    } = this.props;
+    if (currentReservation) {
+      return <th className="resource-table-row available reserved">Varattu</th>;
+    }
+    if (nextReservation) {
+      const availableTime = moment(nextReservation.begin).diff(moment(), 'hours', true).toFixed(1);
+      return (
+        <th className="resource-table-row available">{
```
Comment (lraivio) [inline app/components/resource/ResourcesTableItem.js]: - [ ] Change all the `th` elements in this file to `td`.

File: app/utils/DataUtils.js [hunk 2]
```diff
@@ -100,6 +100,14 @@ function getCaption(item, language = 'fi') {
   return getTranslatedProperty(item, 'caption', language);
 }
 
+function getCurrentReservation(reservations) {
+  const now = moment();
+  return find(
+    reservations, reservation => moment(reservation.begin) < now && now < moment(reservation.end)
+  );
+}
+
```
Comment (lraivio) [inline app/utils/DataUtils.js]: - [ ] One extra empty line.

File: app/utils/__tests__/DataUtils.spec.js [hunk 3]
```diff
@@ -621,4 +624,30 @@ describe('Utils: DataUtils', () => {
       expect(getTranslatedProperty(item, 'name')).to.equal('Finnish name');
     });
   });
+
+  describe('getReservation functions', () => {
```
Comment (lraivio) [inline app/utils/__tests__/DataUtils.spec.js]: - [ ] You could remove this describe block and instead have the `getCurrentReservation` and `getNextReservation` describe blocks on this level each with their own test setup.

File: app/utils/__tests__/DataUtils.spec.js [hunk 4]
```diff
@@ -621,4 +624,30 @@ describe('Utils: DataUtils', () => {
...
+      nextReservation,
+      currentReservation,
+    ];
+    describe('getCurrentReservation', () => {
+      it('returns the current reservation from a list of reservations', () => {
+        expect(getCurrentReservation(unorderedReservations)).to.deep.equal(currentReservation);
+      });
+    });
+    describe('getNextReservation', () => {
+      it('returns the current reservation from a list of reservations', () => {
```
Comment (lraivio) [inline app/utils/__tests__/DataUtils.spec.js]: - [ ] This should say "returns the next reservation...".

File: app/components/resource/ResourcesTable.js [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+import Table from 'react-bootstrap/lib/Table';
+import { connect } from 'react-redux';
+
+import ResourcesTableItem from 'components/resource/ResourcesTableItem';
+import { getCurrentReservation, getNextReservation } from 'utils/DataUtils';
+
+export class UnconnectedResourcesTable extends Component {
+  constructor(props) {
+    super(props);
+    this.renderResourcesTableItem = this.renderResourcesTableItem.bind(this);
```
Comment (lraivio) [inline app/components/resource/ResourcesTable.js]: - [ ] This shouldn't be necessary with the `renderResourcesTableItem`. It's not getting anything from `this`.

File: app/components/resource/ResourcesTable.js [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+            <tr>
+              <th>Tilan nimi</th>
+              <th>Vapaata</th>
+              <th>Varaus</th>
+              <th>Varaaja</th>
+              <th>Kommentit</th>
+            </tr>
+          </thead>
+          <tbody>
+            { resources.map(this.renderResourcesTableItem) }
```
Comment (lraivio) [inline app/components/resource/ResourcesTable.js]: - [ ] This should be without the spaces.

File: app/components/resource/__tests__/ResourcesTable.spec.js [hunk 7]
```diff
@@ -0,0 +1,153 @@
...
+  );
+
+  describe('with resources', () => {
+    let wrapper;
+
+    before(() => {
+      wrapper = getWrapper();
+    });
+
+    it('renders a ul element', () => {
```
Comment (lraivio) [inline app/components/resource/__tests__/ResourcesTable.spec.js]: - [ ] This should read "renders a Table component"

**Comment by lraivio** [comment]
- [ ] There is typo in one of the commit messages: "AdminResourcesPageReductor" -> ""AdminResourcesPageReducer".

File: app/reducers/__tests__/adminResourcesPageReducer.spec.js [hunk 8]
```diff
@@ -0,0 +1,97 @@
...
+
+  describe('handling actions', () => {
+    describe('API.RESOURCES_GET_SUCCESS', () => {
+      const getResourceSuccess = createAction(
+        types.API.RESOURCES_GET_SUCCESS,
+        payload => payload,
+        (payload, meta) => meta
+      );
+
+      it('adds resources ids from action', () => {
```
Comment (lraivio) [inline app/reducers/__tests__/adminResourcesPageReducer.spec.js]: - [ ] You could separate the case where meta.source is 'adminResourcesPage' and the case where meta.source is something else to own `describe` blocks.

File: app/containers/__tests__/AdminResourcesPage.spec.js [hunk 9]
```diff
@@ -0,0 +1,118 @@
...
+
+        it('has correct resources prop', () => {
+          expect(resourcesTable.prop('resources')).to.deep.equal([]);
+        });
+      });
+    });
+  });
+
+  describe('componentDidMount', () => {
+    // describe('if user is not an admin', () => {
```
Comment (lraivio) [inline app/containers/__tests__/AdminResourcesPage.spec.js]: - [ ] Remove commented out code or fix these tests if there was something broken.

File: app/containers/AdminResourcesPage.js [hunk 10]
```diff
@@ -0,0 +1,64 @@
...
+      </DocumentTitle>
+    );
+  }
+}
+
+UnconnectedAdminResourcesPage.propTypes = {
+  actions: PropTypes.object.isRequired,
+  isFetchingResources: PropTypes.bool.isRequired,
+  resources: PropTypes.array.isRequired,
+  isAdmin: PropTypes.bool.isRequired,
```
Comment (lraivio) [inline app/containers/AdminResourcesPage.js]: - [ ] Have the propTypes in alphabetical order.

File: app/components/layout/Navbar.js [hunk 11]
```diff
@@ -34,6 +34,10 @@ class Navbar extends Component {
             <MenuItem>Omat varaukset</MenuItem>
           </LinkContainer>
           <MenuItem divider />
+          <LinkContainer to="/admin-resources">
+            <MenuItem>Omat tilat (Admin)</MenuItem>
```
Comment (lraivio) [inline app/components/layout/Navbar.js]: - [ ] This could be just "Omat tilat" without the "(Admin)"

**Comment by lraivio** [comment]
- [ ] Only preliminary reservations have the reserverName property. 
So if that does not exist you have to use `reservation.user.displayName` or `reservation.user.email`.

**Comment by lraivio** [comment]
- [ ] The resource names should be links to that resource's page.

**Comment by lraivio** [comment]
- [ ] For me it doesn't show anything in the available time if there is no reservations for today.

**Comment by lraivio** [comment]
- [ ] The available time or "reserved" text could be colored as in the mockups.

**Comment by YoYuUm** [comment]
@lraivio, So you say if the next reservation is another day it should show information of available time till next reservation (in hours, days?), and what about the lenght of the reservation (now it from time to time, but what if it is another day, should it show perhaps the date too?).

Now it doesn't show the next reservation because I'm getting the resources with the start and end parameters with the begining and end of the day. If we want to show more days, we should change the end date to something ahead in the future. How many days should I add ?

**Comment by lraivio** [comment]
@YoYuUm:
If there are no reservations today but the resource is open (it has opening hours) then the available time in "Vapaata" column should be the time from now till the closing hour of the resource. So how many hours the user could reserve immediately today.
In the reservation columns ("Varaus", "Varaajat" and "Kommentit") nothing needs to be shown if there are no reservations today.

**Pull request merged**